src/nvim/globals.h

```diff
@@ -746,6 +746,9 @@ EXTERN bool enc_utf8 INIT(= false);             /* UTF-8 encoded Unicode */
 EXTERN int enc_latin1like INIT(= TRUE);         /* 'encoding' is latin1 comp. */
 EXTERN int has_mbyte INIT(= 0);                 /* any multi-byte encoding */
 
+// 'encoding' can only be changed in the startup scripts
+EXTERN bool enc_mutable INIT(= true);
```

Comment:
The problem is that "at VimEnter" might be to late: both BufEnter and VimEnter are executed _after_ the TUI is initialized, yes it is a contrived example, but then it is possible to do
`au BufEnter * set encoding=something`
and possibly break the TUI input.
Why not just rename `enc_mutable` to `startupscripts_loaded` or something?